### PR TITLE
Create filter slice:

### PR DESCRIPTION
**Pull request by G1Lroy**
-move filters logic to new slice
-change filtration logic (realtime to click on button)

Suggestions:
-move other slices to  separate folders, cause interfases grows
  slices {
    folderName {
        store
        model
    }
    folderName2{
        //
        //
    }
  }

### CODE REVIEW

File: src/components/ProductFilters.tsx [hunk 1]
```diff
@@ -1,103 +1,117 @@
...
-];
 export type ProductFiltersProps = {};
 
 const ProductFilters: FC<ProductFiltersProps> = () => {
-  const filter = useTypedSelector((state) => state.productList.productFilters);
-  const [maxPrice, setMaxPrice] = useState<number>(filter.price.max);
   const dispatch = useAppDispatch();
+  const {filterOptions} = useTypedSelector((state) => state.productFilters);
+  const [activeButton, setActiveButton] = useState<boolean>(false);
+  const [maxPrice, setMaxPrice] = useState<number>(100);
```
Comment (VladislavMinchuk) [inline src/components/ProductFilters.tsx:28]: move min-max to static data and get values from there (useState unnecessary)

File: src/components/ProductFilters.tsx [hunk 2]
```diff
@@ -1,103 +1,117 @@
...
+  const [minPrice, setMinPrice] = useState<number>(1);
 
-  const handleChangeSort = (e: ChangeEvent<HTMLSelectElement>) => {
-    dispatch(sortProductsByType(e.target.value));
-  };
-  const handleChandeMaxPrice = (e: ChangeEvent<HTMLInputElement>) => {
-    const currMaxPriceValue = Number(e.target.value);
-    if (currMaxPriceValue < 1) return;
-    setMaxPrice(currMaxPriceValue);
+  const handlePriceUpdate = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (VladislavMinchuk) [inline src/components/ProductFilters.tsx:31]: we can get value from 'handleSliderMouseUp'. 
'setMaxPrice' calls component re-render every time. onChange event calling too often :)

File: src/components/ProductFilters.tsx [hunk 3]
```diff
@@ -1,103 +1,117 @@
...
+    const valueMax = Number(e.target.value);
+    setMaxPrice(valueMax);
+    // setMinPrice(value)
+    setActiveButton(true);
```
Comment (VladislavMinchuk) [inline src/components/ProductFilters.tsx]: Suggestion: maybe we can make it with 'useEffect' that watching by 'filterOptions' object (set 'activeButton' inside hook). I'm not sure.
And remove setActiveButton from each handler.

File: src/components/ProductFilters.tsx [hunk 4]
```diff
@@ -1,103 +1,117 @@
...
-    dispatch(setGloablLoading(true));
-
-    setTimeout(() => {
-      dispatch(filterProductsByPrice(maxPrice));
-      dispatch(setGloablLoading(false));
-    }, 500);
+  const handleSliderMouseUp = () => {
+    dispatch(setPriceFilter({ max: maxPrice, min: minPrice }));
   };
   const handleUpdateCheckBox = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (VladislavMinchuk) [inline src/components/ProductFilters.tsx:39]: We can improve 'setSizeFilter' reducer. Event object has 'checked' boolean field (e.target.checked).
Use is for 'setSizeFilter' reducer.
Based on 'checked' field from input we can call 'push' or 'filter' without 'includes' condition.

File: src/store/slices/productFilters/index.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+export const productFiltersSlice = createSlice({
+  name: "productFilters",
+  initialState,
+  reducers: {
+    setPriceFilter(state, action: PayloadAction<{ min: number; max: number }>) {
+      state.filterOptions.price = action.payload;
+    },
+    setSizeFilter(state, action: PayloadAction<string>) {
```
Comment (VladislavMinchuk) [inline src/store/slices/productFilters/index.ts]: Based on 'checked' field from input we can call 'push' or 'filter' without 'includes' condition.

File: src/components/ProductFilters.tsx [hunk 6]
```diff
@@ -1,103 +1,117 @@
...
-  useEffect(() => {
-    // console.log(filter.sizes.chosen);
-    // console.log(filter.sizes.available);
-    dispatch(filterProductsBySize());
-  }, [filter.sizes.chosen]);
+  const aplyFilters = () => {
+    console.log(filterOptions);
+    setActiveButton(false);
+    dispatch(setGloablLoading(true));
+    setTimeout(() => dispatch(setGloablLoading(false)), 300);
```
Comment (VladislavMinchuk) [inline src/components/ProductFilters.tsx]: Incorrect works: loader displaying but list already updated. User shouldn't see loader and new updated list at the same time.
For ex: update setTimeout value to 2000 and try it

File: src/components/ProductFilters.tsx (hunk 2 above)
Comment (G1Lroy) [reply src/components/ProductFilters.tsx:31]: We cannot access the input value in this handler; it processes events related to the mouse, for example, you can find out the position of the mouse on the screen, etc.  
We can move the dispatch to the oNchange handler and delete oNmouse, but then we will contact the store for every change.

File: src/components/ProductFilters.tsx (hunk 3 above)
Comment (G1Lroy) [reply src/components/ProductFilters.tsx]: Done!

File: src/components/ProductFilters.tsx (hunk 4 above)
Comment (G1Lroy) [reply src/components/ProductFilters.tsx:39]: done

File: src/components/ProductFilters.tsx (hunk 6 above)
Comment (G1Lroy) [reply src/components/ProductFilters.tsx]: done

File: src/components/ProductFilters.tsx (hunk 1 above)
Comment (G1Lroy) [reply src/components/ProductFilters.tsx:28]: I don't fully understand the remark.
These states store the value of inputs (in the future I would like to add a pointer for the minimum price). Then, with a mouse up, the value is transferred to the dispatcher and stored in the store.
You may have thought about these values that we specify for the maximum and minimum input values. But I hardcoded them in the <Form.Range component
             min={1}
             max={100}

**Pull request merged**